Remove a lot of mentions of JavaScript versions (#19448)
mdn/content: files/en-us/web/javascript/reference/trailing_commas/index.md

### Trailing commas in JSON

> As JSON is based on JavaScript's syntax prior to ES5, **trailing commas are not allowed in JSON**.

Reviewer: Nit (matter of taste):
```suggestion
As JSON is based on JavaScript's syntax before ES5, _trailing commas are not allowed in JSON_.
```
